src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java

```diff
@@ -0,0 +1,163 @@
+
+        if (tileEntity instanceof TileEntityWareHouse)
+        {
+            final TileEntityWareHouse wareHouse = (TileEntityWareHouse) tileEntity;
+
+            return wareHouse.isInHut(requestToCheck.getRequest());
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public List<IToken> attemptResolve(
+                                        @NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request)
+    {
+        final TileEntity tileEntity = manager.getColony().getWorld().getTileEntity(getLocation().getInDimensionLocation());
+
+        if (tileEntity instanceof TileEntityWareHouse)
+        {
+            final TileEntityWareHouse wareHouse = (TileEntityWareHouse) tileEntity;
+            final BlockPos pos = wareHouse.getPositionOfChestWithItemStack(request.getRequest());
+
+            request.setResult(request.getRequest().copy());
+            return Lists.newArrayList(manager.createRequest(new WarehouseChestDeliveryRequester(this, manager.getFactoryController().getNewInstance(UUID.randomUUID(),
+              new TypeToken<IToken>() {}), manager.getFactoryController().getNewInstance(pos, new TypeToken<ILocation>() {}), request.getToken()),
+              new Delivery(manager.getFactoryController().getNewInstance(
+                pos,
+                new TypeToken<ILocation>() {}), request.getRequester().getDeliveryLocation(), request.getRequest())));
+        }
+
+        return Lists.newArrayList();
+    }
+
+    @Nullable
+    @Override
+    public void resolve(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request) throws RuntimeException
+    {
+        //Noop delivery has been completed
+    }
+
+    @Nullable
+    @Override
+    public IRequest getFollowupRequestForCompletion(
+                                                     @NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> completedRequest)
+    {
+        //No followup needed.
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public IRequest onParentCancelled(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request) throws IllegalArgumentException
+    {
+        //TODO Release the stack from the warehouse.
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public void onResolvingOverruled(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request) throws IllegalArgumentException
+    {
+        //TODO Release the stack from the warehouse.
+    }
+
+    private final class WarehouseChestDeliveryRequester implements IRequester
+    {
+
+        private final WarehouseRequestResolver warehouseRequestResolver;
+        private final IToken                   id;
+        private final ILocation                location;
+
+        private final IToken itemStackRequestToken;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 33 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)